src/samplereader/ADTSSampleReader.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ADTSSampleReader.h"
+
+ADTSSampleReader::ADTSSampleReader(AP4_ByteStream* input, AP4_UI32 streamId)
+  : ADTSReader{input},
+    m_streamId{streamId},
+    m_adByteStream{dynamic_cast<AdaptiveByteStream*>(input)} {};
+
+AP4_Result ADTSSampleReader::Start(bool& bStarted)
+{
+  bStarted = false;
+  if (m_started)
+    return AP4_SUCCESS;
+  bStarted = true;
+  m_started = true;
+  return ReadSample();
+}
+
+AP4_Result ADTSSampleReader::ReadSample()
+{
+  if (ReadPacket())
+  {
+    if (uint64_t pts = GetPts() == PTS_UNSET)
```

Comment:
```suggestion
   uint64_t pts {GetPts()};
   if (pts == PTS_UNSET)
```
define variable outside the condition